lms/djangoapps/grades/new/course_grade.py

```diff
@@ -83,15 +119,33 @@ def percent(self):
         """
         Returns a rounded percent from the overall grade.
         """
+        if self._percent is not None:
+            return self._percent
         return self._calc_percent(self.grade_value)
```

Comment:
Don't you want to set `self._percent` to the result of this call?  You can simplify this code and make it consistent with how we handle this type of lazy computation elsewhere by changing this to:

```
@property
def percent(self):
if self._percent is None:
    self._percent = self._calc_percent(self.grade_value)
return self._percent
```

And then you can remove the `setter`.  Alternatively, don't do any of the above and just use the `@lazy` decorator instead of `@property`.